flash/graph/classification/input.py

```diff
@@ -11,39 +11,41 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from typing import Any, Mapping
+from typing import Any, Dict, Mapping, Optional
 
 from torch.utils.data import Dataset
 
-from flash.core.data.data_module import DatasetInput
-from flash.core.data.io.classification_input import ClassificationInputMixin, ClassificationState
-from flash.core.data.io.input import DataKeys
+from flash.core.data.io.classification_input import ClassificationInputMixin
+from flash.core.data.io.input import DataKeys, Input
+from flash.core.data.utilities.samples import to_sample
 from flash.core.utilities.imports import _GRAPH_AVAILABLE, requires
 
 if _GRAPH_AVAILABLE:
     from torch_geometric.data import Data
-    from torch_geometric.data import Dataset as TorchGeometricDataset
-    from torch_geometric.data import InMemoryDataset
 
 
-class GraphClassificationDatasetInput(DatasetInput, ClassificationInputMixin):
+def _get_num_features(sample: Dict[str, Any]) -> Optional[int]:
+    """Get the number of features per node in the given dataset."""
+    data = sample[DataKeys.INPUT]
+    data = data[0] if isinstance(data, tuple) else data
+    return getattr(data, "num_node_features", None)
+
+
+class GraphClassificationDatasetInput(Input, ClassificationInputMixin):
     @requires("graph")
     def load_data(self, dataset: Dataset) -> Dataset:
         if not self.predicting:
-            if isinstance(dataset, TorchGeometricDataset):
-                self.num_features = dataset.num_features
-
-                if isinstance(dataset, InMemoryDataset):
-                    self.load_target_metadata([sample.y for sample in dataset])
-                else:
-                    self.num_classes = dataset.num_classes
-                    self.set_state(ClassificationState(self.labels, self.num_classes))
+            # TODO: This may be slow
+            samples = [self.load_sample(sample) for sample in dataset]
```

Comment:
Are you sure about this? I won't do this. You could use only the first sample maybe.

Reply:
Can't use the first sample as it won't contain all of the labels / information we need to properly resolve the targets. In #1123 I'm adding a way to allow users to provide their `TargetFormatter` when inference is not possible. This would then be required if you want to use `from_datasets` and your targets need to be formatted (e.g. you won't need to provide anything if your targets are just integers).

Reply:
Ok, you now just have to provide a `TargetFormatter` object if you want the targets to be handled in a specific way